tests/integration/test_sqs.py

```diff
@@ -2487,6 +2487,36 @@ def test_purge_queue_deletes_delayed_messages(self, sqs_create_queue, aws_client
         receive_result = aws_client.sqs.receive_message(QueueUrl=queue_url, WaitTimeSeconds=1)
         assert "Messages" not in receive_result.keys()
 
+    @pytest.mark.aws_validated
+    def test_purge_queue_clears_fifo_deduplication_cache(self, sqs_create_queue, aws_client):
+        fifo_queue_name = f"test-queue-{short_uid()}.fifo"
+        queue_url = sqs_create_queue(QueueName=fifo_queue_name, Attributes={"FifoQueue": "true"})
+        dedup_id = f"fifo_dedup-{short_uid()}"
+        group_id = f"fifo_group-{short_uid()}"
+
+        aws_client.sqs.send_message(
+            QueueUrl=queue_url,
+            MessageBody="message-1",
+            MessageGroupId=group_id,
+            MessageDeduplicationId=dedup_id,
+        )
+
+        aws_client.sqs.purge_queue(QueueUrl=queue_url)
+
+        aws_client.sqs.send_message(
```

Comment:
In the original reproduction template of your issue https://github.com/localstack/localstack/issues/8211 there was an additional step here before the purge (a receive message call) - is this necessary here?

Reply:
It isn't necessary. The issue reproduction step was just to illustrate that the message was pushed previously & could be retrieved. That is covered by other tests.